services/workflows-service/src/data-analytics/data-analytics.service.ts

```diff
@@ -46,16 +50,120 @@ export class DataAnalyticsService {
         });
     }
 
-    // Used for exhaustive check
-    inlineRule satisfies never;
-
     this.logger.error(`No evaluation function found`, {
       inlineRule,
     });
 
     throw new Error(`No evaluation function found for rule name: ${(inlineRule as InlineRule).id}`);
   }
 
+  async checkMerchantOngoingAlert(
+    businessReport: BusinessReport,
+    {
+      increaseRiskScorePercentage,
+      increaseRiskScore,
+      maxRiskScoreThreshold,
+    }: CheckRiskScoreOptions,
+    alertSeverity: AlertSeverity,
+  ) {
+    const { report, businessId, projectId } = businessReport;
+
+    if (
+      !(
+        report as {
+          data: {
+            previousReport?: unknown;
+          };
+        }
+      ).data.previousReport
+    ) {
+      return;
+    }
+
+    const {
+      data: {
+        summary: { riskScore: currentRiskScore },
+        previousReport: {
+          summary: { riskScore: previousRiskScore },
+          reportType: previousReportType,
+        },
+      },
+    } = report as {
+      data: {
+        summary: {
+          riskScore: number;
+        };
+        previousReport: {
+          summary: {
+            riskScore: number;
+          };
+          reportType: BusinessReportType;
+        };
+      };
+    };
+
+    if (previousReportType !== BusinessReportType.ONGOING_MERCHANT_REPORT_T1) {
+      this.logger.warn(`Previous report type is not ONGOING_MERCHANT_REPORT_T1`);
+
+      return;
+    }
+
+    if (currentRiskScore < previousRiskScore) {
+      return;
+    }
+
+    if (!(maxRiskScoreThreshold || increaseRiskScore || increaseRiskScorePercentage)) {
```

Comment:
Negating an entire expression is more open to human error and missing it in my opinion.
```suggestion
    if (!maxRiskScoreThreshold || !increaseRiskScore || !increaseRiskScorePercentage) {
```